src/figuresClasses.ts

```diff
@@ -1,11 +1,95 @@
-export interface Figure {}
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  public shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
```

Comment:
Since TypeScript automatically creates class properties from constructor parameters when using access modifiers, you don't need to declare them explicitly. You can remove the property declarations on lines 11, 15, 16, 17, 51, 55, 70, 74, and 75.